app/src/main/kotlin/edu/stanford/bdh/engagehf/MainActivity.kt

```diff
@@ -216,6 +216,11 @@ class MainActivity : FragmentActivity() {
                         is AppNavigationEvent.AppScreen -> navHostController.navigate(Routes.AppScreen)
```

Comment:
We need the same clearing back stack mechanism for this event as well, here we need to clear the back stack when we receive the sign in / sign up success event in the view model, e.g. `AppNavigationEvent.AppScreen(clearBackStack = true)`, and with the flag false when calculating the start destination. Currently we are able to go back after the login which we should avoid.

With this, we can create the following extension function on nav host controller

```kotlin
private fun NavHostController.navigateTo(route: Routes, clearBackStack: Boolean = false) {
    navigate(route) {
        if (clearBackStack) {
            popUpTo(graph.startDestinationId) { inclusive = true }
            launchSingleTop = true
        }
    }
}
```

And reuse for both `AppScreen` and `OnboardingScreen` event

Reply:
yes, it makes total sense that we also introduce this with the AppScreen